### PR TITLE
Improvements to springsalad specific properties panel and shape representation

### PR DESCRIPTION
**Pull request by danv61**
selected shape highlight - site oval shape and link line shape
interactive selection updates between the sites / links table and the corresponding shapes - works both ways

### CODE REVIEW

File: vcell-client/src/main/java/cbit/vcell/mapping/gui/LinkSpecsTableModel.java [hunk 1]
```diff
@@ -260,6 +259,24 @@ protected List<MolecularInternalLinkSpec> computeData() {
...
             return null;
         }
+
+        Collections.sort(allParameterList, new Comparator<MolecularInternalLinkSpec>() {
+            @Override
+            public int compare(MolecularInternalLinkSpec mils1, MolecularInternalLinkSpec mils2) {
+                SpeciesContextSpec scs = mils1.getSpeciesContextSpec();
+                if(fieldSpeciesContextSpec != scs) {
+                    throw new RuntimeException("SpeciesContextSpec inconsistent.");
+                }
```
Comment (jcschaff) [inline vcell-client/src/main/java/cbit/vcell/mapping/gui/LinkSpecsTableModel.java:269]: do you need to verify that mils2 also points to the correct SpeciesContextSpec?  it is okay if you think that is overkill.

File: vcell-client/src/main/java/cbit/vcell/mapping/gui/MolecularStructuresPropertiesPanel.java [hunk 2]
```diff
@@ -140,11 +166,39 @@ public boolean isViewSingleRow() {
                 }
             };
             shapePanel.addMouseMotionListener(new MouseMotionAdapter() {
+                @Override
                 public void mouseMoved(MouseEvent e) {
+                    // this will fire if the mouse moves over the panel, we may need it for drag and drop
+                    super.mouseMoved(e);
+//                    Point overWhat = e.getPoint();
```
Comment (jcschaff) [inline vcell-client/src/main/java/cbit/vcell/mapping/gui/MolecularStructuresPropertiesPanel.java:173]: remove when you are done with this code.

File: vcell-core/src/main/java/cbit/vcell/mapping/SpeciesContextSpec.java [hunk 3]
```diff
@@ -70,8 +70,14 @@ public class SpeciesContextSpec implements Matchable, ScopedSymbolTable, Seriali
...
     private static final String PROPERTY_NAME_WELL_MIXED = "wellMixed";
     private static final String PROPERTY_NAME_FORCECONTINUOUS = "forceContinuous";
-    public static final String PROPERTY_NAME_SITE_ATTRIBUTE = "SiteAttributes";     // springsalad
-    public static final String PROPERTY_NAME_LINK_LENGTH = "LinkLength";            // springsalad
+    // springsalad group
+    public static final String PROPERTY_NAME_SITE_ATTRIBUTE = "SiteAttributes";
+    public static final String PROPERTY_NAME_LINK_LENGTH = "LinkLength";
+    public static final String PROPERTY_NAME_SITE_SELECTED_IN_TABLE = "SiteSelectedInTable";
+    public static final String PROPERTY_NAME_LINK_SELECTED_IN_TABLE = "LinkSelectedInTable";
+    public static final String PROPERTY_NAME_SITE_SELECTED_IN_SHAPE = "SiteSelectedInShape";
```
Comment (jcschaff) [inline vcell-core/src/main/java/cbit/vcell/mapping/SpeciesContextSpec.java:78]: for these 4 new bound properties, make sure you un-listen and re-listen where appropriate to (1) keep the UI responsive even after add/remove of objects and (2) prevent memory leaks and invisible components trying to update themselves.

**Review by jcschaff — APPROVED** [review]
Please look at comments and address them as appropriate (if at all).  Looks very nice and fun Graphics code.

File: vcell-client/src/main/java/cbit/vcell/mapping/gui/LinkSpecsTableModel.java (hunk 1 above)
Comment (danv61) [reply vcell-client/src/main/java/cbit/vcell/mapping/gui/LinkSpecsTableModel.java:269]: Actually even the test for mils1 is overkill, I'm only trying to catch bugs early during development. Condition should bever happen in a bug-free code.

File: vcell-client/src/main/java/cbit/vcell/mapping/gui/MolecularStructuresPropertiesPanel.java (hunk 2 above)
Comment (danv61) [reply vcell-client/src/main/java/cbit/vcell/mapping/gui/MolecularStructuresPropertiesPanel.java:173]: I'll probably remove the whole mouse motion listener eventually, unless it's needed for the drag and drop implementation

**Pull request merged**